benches/distributions.rs

```diff
@@ -46,27 +46,49 @@ macro_rules! distr_float {
                     let x: $ty = distr.sample(&mut rng);
                     accum += x;
                 }
-                black_box(accum);
+                accum
             });
             b.bytes = size_of::<$ty>() as u64 * ::RAND_BENCH_N;
         }
     }
 }
 
-macro_rules! distr {
-    ($fnn:ident, $ty:ty, $distr:expr) => {
+macro_rules! distr_bool {
+    ($fnn:ident, $distr:expr) => {
         #[bench]
         fn $fnn(b: &mut Bencher) {
             let mut rng = XorShiftRng::from_entropy();
             let distr = $distr;
 
             b.iter(|| {
+                let mut accum = true;
                 for _ in 0..::RAND_BENCH_N {
-                    let x: $ty = distr.sample(&mut rng);
-                    black_box(x);
+                    let x: bool = distr.sample(&mut rng);
+                    accum ^= x;
                 }
+                accum
             });
-            b.bytes = size_of::<$ty>() as u64 * ::RAND_BENCH_N;
+            b.bytes = size_of::<bool>() as u64 * ::RAND_BENCH_N;
+        }
+    }
+}
+
+macro_rules! distr_char {
+    ($fnn:ident, $distr:expr) => {
+        #[bench]
+        fn $fnn(b: &mut Bencher) {
+            let mut rng = XorShiftRng::from_entropy();
+            let distr = $distr;
+
+            b.iter(|| {
+                let mut accum = 0u32;
+                for _ in 0..::RAND_BENCH_N {
+                    let x: char = distr.sample(&mut rng);
+                    accum = accum.wrapping_add(x as u32);
```

Comment:
if this is the only difference, I think you could just do the same for booleans, though doesn't really matter